packages/salesforcedx-vscode-core/src/metadataSupport/metadataDocumentationService.ts

```diff
@@ -0,0 +1,720 @@
+      CustomField: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/customfield.htm',
+      CustomFieldTranslation:
+        'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_customobjecttranslation.htm',
+      CustomSite: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_sites.htm',
+      CustomTab: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_tab.htm',
+      DocumentFolder: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_document.htm',
+      EmailFolder: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_emailtemplate.htm',
+      Flow: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_visual_workflow.htm',
+      MatchingRules: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_matchingrule.htm',
+      ReportFolder: 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_report.htm'
+    };
+
+    // Return known URL if available
+    if (knownUrls[metadataType]) {
+      return [knownUrls[metadataType]];
+    }
+
+    // Try multiple URL patterns for unknown metadata types
+    const baseUrl = 'https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta';
+    const typeNameLower = metadataType.toLowerCase();
+
+    // Generate potential URLs to try (in order of preference)
+    const potentialUrls = [
+      `${baseUrl}/meta_${typeNameLower}.htm`,
+      `${baseUrl}/meta_${typeNameLower}s.htm`, // plural form
+      `${baseUrl}/meta_${typeNameLower}es.htm`, // plural form with es suffix
+      `${baseUrl}/meta_${metadataType.replace(/([A-Z])/g, '_$1').toLowerCase()}.htm` // snake_case
+    ];
+
+    return potentialUrls;
+  }
+
+  /**
+   * Get hardcoded field definitions for common metadata fields
+   */
+  private getFieldDefinitions(metadataType: string, fieldName: string): MetadataFieldDocumentation | null {
+    const commonFields: Record<string, MetadataFieldDocumentation> = {
+      // Common fields across all metadata types
+      fullName: {
+        name: 'fullName',
+        type: 'string',
+        description: 'The unique name of the metadata component. This is the API name used to reference the component.',
+        required: false
+      },
+      description: {
+        name: 'description',
+        type: 'string',
+        description: 'A description of the metadata component. This field is optional and provides additional context.',
+        required: false
+      },
+      masterLabel: {
+        name: 'masterLabel',
+        type: 'string',
+        description: 'The display label for the metadata component. This is what users see in the Salesforce UI.',
+        required: false
+      }
+    };
+
+    // Metadata type specific fields
+    const typeSpecificFields: Record<string, Record<string, MetadataFieldDocumentation>> = {
+      CustomObject: {
```

Comment:
again, what made you choose these fields instead of doing them all programatically?  And where's the code the scrapes the docs for each type to get its fields?